python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+"""
+Kriging: metamodel with continuous and categorical variables
+============================================================
+"""
+# %%
+# We consider here the surrogate modeling of an analytical function characterized by
+# continuous and categorical variables
+#
+
+# %%
+import openturns as ot
+import openturns.experimental as otexp
+import numpy as np
+import matplotlib.pyplot as plt
+
+ot.RandomGenerator.SetSeed(5)
+
+# %%
+# We first show the advantage of modeling the various levels of a mixed
+# continuous / categorical function through a single surrogate model
+# on a simple test-case, defined below.
+
+
+# %%
+def illustrativeFunc(inp):
```

Comment:
This function name  could be chosen in every example in the doc. What is the specific purpose of this particular function?

Reply:
Is that function used in the paper (Zhang et al., 2020)?

Reply:
The two test functions are from the PhD thesis of J. Pelamatti. A reference has been added.